hoomd/state.py

```diff
@@ -55,52 +75,58 @@ def __init__(self, simulation, snapshot):
         else:
             self._cpp_sys_def = _hoomd.SystemDefinition(
                 snapshot._cpp_obj, simulation.device._cpp_exec_conf)
+
+        # Necessary for local snapshot API. This is used to ensure two local
+        # snapshots are not contexted at once.
+        self._in_context_manager = False
+
+        # provides a cache of C++ group objects of the form {type(filter):
+        # {filter: C++ group}}. The first layer is to prevent user created
+        # filters with poorly implemented __hash__ and __eq__ from causing cache
+        # errors.
         self._groups = defaultdict(dict)
 
     @property
     def snapshot(self):
+        """hoomd.Snapshot: All data of a simulation's current microstate.
+
+        `State.snapshot` should be used when all of a simulation's state
+        information is desired in a single object. When accessed, data across
+        all MPI ranks and from GPUs is gathered on the root MPI rank's memory.
+        When accessing data in MPI simulations then it is recommended to use a
+        ``if snapshot.exists:`` conditional to prevent attempting to access data
+        on a non-root rank.
+
+        This property can be set to the state using an entirely new
+        `hoomd.Snapshot` object.  Example use cases in which a simulation's
```

Comment:
```suggestion
        This property can be set to replace the system state with the given
        `hoomd.Snapshot` object.  Example use cases in which a simulation's
```
"entirely new" here is awkward. I could be an existing snapshot that has been modified.